pkg/services/ngalert/api/api_alertmanager_base.go

```diff
@@ -110,6 +88,20 @@ func (base AlertmanagerApiBase) RouteGetAlertingConfig(c *models.ReqContext) res
 	return response.Error(http.StatusNotImplemented, "", nil)
 }
 
+func (base AlertmanagerApiBase) RouteGetSilence(c *models.ReqContext) response.Response {
+	silenceId := c.Params(":SilenceId")
+	base.log.Info("RouteGetSilence: ", "SilenceId", silenceId)
+	recipient := c.Params(":Recipient")
+	base.log.Info("RouteGetSilence: ", "Recipient", recipient)
```

Comment:
A general comment for the entire file - ideally it should be a single log line to be able to co-relate the multiple fields which are in different logs lines right now. There could come multiple such log lines in between these 2 log lines and it can become nearly impossible to know which silence id was for which recipient.

Reply:
Good point but this is not actually used; it's auto-generated and it's for demoing how parameters can get accessed.
The actual implementation is [this](https://github.com/grafana/grafana/blob/a06b3335f2482bba9ab6136461d1282dddec2785/pkg/services/ngalert/api/api_alertmanager.go).